BlueUtils/SpecialWires.bsv

```diff
@@ -0,0 +1,65 @@
+package SpecialWires; 
+
+export mkDWireOR;
+export mkWireOP;
+
+import List :: *;
+
+// DWire where all writes are OR'd together
+// Useful when setting fields of a Struct in
+// different rules
+module mkDWireOR #( Integer n
+                  , a_bits init
+  ) (Array #(Wire #(a_bits))) provisos (
+    Bits #(a_bits, bit_size));
+
+  let v <- mkWireOP (n, mkDWire (init), \| );
+  return v;
+endmodule
+
+// Use this for OPs on Bits
+// Force the op to be on Bit #(w) so that when passing an op on Bits the compiler won't
+// complain about the op being of ambiguous type
+module mkWireOP #( Integer n
+                 , function module #(Wire #(a_bits)) wire_module
+                 , function Bit #(bit_size) op (Bit #(bit_size) arg1, Bit #(bit_size) arg2)
+  ) (Array #(Wire #(a_bits))) provisos (
+    Bits #(a_bits, bit_size));
+
+    let v <- mkWireOP_Core (n, wire_module, op);
+    return v;
+endmodule
+
+// Only useful in rare cases when using some strange 'op'
+// Most generic version, does all of the work for the wrappers
+// To use with a generic Bits op (eg \|), must define helper fn. Eg:
+// function Bit #(n) or_fn (Bit #(n) arg1, Bit #(n) arg2) = arg1 | arg2;
+module mkWireOP_Core #( Integer n
+                      , function module #(Wire #(a_bits)) wire_module
+                      , function b_bits op (c_bits arg1, d_bits arg2)
+  ) (Array #(Wire #(a_bits))) provisos (
+    Bits #(a_bits, bit_size)
+  , Bits #(b_bits, bit_size)
+  , Bits #(c_bits, bit_size)
+  , Bits #(d_bits, bit_size));
+
+  if (n<0) error(quote("mkWireOP") + " cannot have a negative number of ports");
```

Comment:
We probably want "at least one" interface? What do you think @JonasAlaif ?

Reply:
Looks like `mkCReg` is happy with a zero sized array so let's keep it like this as well